app/edge/components/edit-edge-stack-form/editEdgeStackFormController.js

```diff
@@ -57,6 +57,15 @@ export class EditEdgeStackFormController {
   checkEndpointTypes(groups) {
     const edgeGroups = groups.map((id) => this.edgeGroups.find((e) => e.Id === id));
     this.state.endpointTypes = edgeGroups.flatMap((group) => group.EndpointTypes);
+    this.selectValidDeploymentType();
+  }
+
+  selectValidDeploymentType() {
+    const validTypes = getValidEditorTypes(this.state.endpointTypes);
+
+    if (!validTypes.includes(this.model.DeploymentType)) {
+      this.onChangeDeploymentType(validTypes[0]);
```

Comment:
do we need to check the length of `validTypes`?

Reply:
Luckily not - validTypes[0] can be undefined as an argument to the `onChangeDeploymentType` function, setting formValues.DeploymentType to undefined if there are no available options